compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1454,6 +1463,22 @@ impl NodeInterner {
                 self.traits.get(&trait_impl.trait_id).map(|trait_| trait_.location)
             })
     }
+
+    fn try_resolve_trait_method_declaration(&self, location: Location) -> Option<Location> {
+        self.func_meta
+            .iter()
+            .find(|(_, func_meta)| func_meta.location.contains(&location))
+            .and_then(|(func_id, _func_meta)| {
+                let (_, trait_id) = self.get_function_trait(func_id)?;
+
+                self.traits
+                    .get(&trait_id)?
+                    .methods
+                    .iter()
+                    .find(|method| method.name.0.contents == self.function_name(func_id))
+                    .map(|method| method.location)
```

Comment:
Working around the formatting
```suggestion
                let methods = self.traits.get(&trait_id)?.methods.iter();
                let method = methods.find(|method| method.name.0.contents == self.function_name(func_id));
                method.map(|method| method.location)
```